Testing in local appengine environment causes non-test-failure errors

I encountered two issues testing locally with google appengine instead of requests lib:

1. Locally invoking urlfetch.fetch raises `*** AssertionError: No api proxy found for service "urlfetch"`
2. https://github.com/EasyPost/easypost-python/blob/8af4e9d7fa5d6f71d2cc4625ef546bab489d76e2/tests/test_shipment.py#L177 assumes that easypost.requests_session exists, which is only true when using requests lib.

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -10,6 +10,25 @@ TEST_API_KEY = os.environ['TEST_API_KEY']
 PROD_API_KEY = os.environ['PROD_API_KEY']
 
 
+def pytest_sessionstart(session):
+    # this is for local unit testing with google appengine, otherwise you get a
+    # 'No api proxy found for service "urlfetch"' response
+    try:
+        from google.appengine.ext import testbed
+        session.appengine_testbed = testbed.Testbed()
+        session.appengine_testbed.activate()
+        session.appengine_testbed.init_urlfetch_stub()
+    except ImportError:
+        # if the import fails then we're not using the appengine sdk, so just
+        # keep going without initializing the testbed stub
+        pass
+
+
+def pytest_sessionfinish(session, exitstatus):
+    if hasattr(session, "appengine_testbed"):
+        session.appengine_testbed.deactivate()
+
+
 # this fixture is auto-loaded by all tests; it sets up the api key
 @pytest.yield_fixture(autouse=True)
 def setup_api_key():

--- a/tests/test_shipment.py
+++ b/tests/test_shipment.py
@@ -174,7 +174,8 @@ def test_rerate(vcr):
         # we only rerate on get_rates calls for shipments made at least 60 seconds ago
         time.sleep(61)
 
-    easypost.requests_session.close()
+    if hasattr(easypost, 'requests_session'):  # the urlfetch code has no session obj
+        easypost.requests_session.close()
 
     shipment.regenerate_rates()
 

